---
title: "Bioporongs"
output: rmarkdown::pdf_document
vignette: >
  %\VignetteIndexEntry{Bioporongs}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(Bioporongs)
```

# Description of the Packages

This R package provides tools to simulate and analyze nucleic acid sequences, allowing users to create a random DNA sequence, convert it to RNA, segment it into codons, and finally translate these codons into an amino acid sequence. By starting with the random DNA generator, users can quickly obtain a sequence that can be converted to RNA, making it compatible for further processing. The RNA sequence can then be divided into codons, with customizable start points, to reflect different reading frames. Finally, the codon sequence is translated into its corresponding amino acid sequence, facilitating analysis of protein-coding regions and supporting applications in genetic studies, bioinformatics, and sequence analysis tasks.

### Function 1 - Random_DNA_seq
This function creates a random DNA sequence with a specific *sequence length*.

```{r}
#EXAMPLE
Random_DNA_seq(10)
```

This is the sequence that can then be used in the following function.


### Function 2 - dna2rna
This function takes a given *dna_seq* and returns its corresponding RNA sequence

```{r}
#EXAMPLE
dna2rna("ATGCGTACGTTAGCCGTAGCTTGAATGCTAGC")
```


### Function 3 - get_codons

This function divide a biological nucleic-acid sequences (DNA or RNA) and returns a list of the codons. The starting point can be modified with the argument *start*.

```{r}
#EXAMPLE
get_codons("ATGCGTACGTTAGCCGTAGCTTGAATGCTAGC", start = 3)
```

## Function 4 - aminoacid_sequence

This function gets the sequence of *codons* and returns the amino acid sequence translation

```{r}
#EXAMPLE
aminoacid_sequence(c("AUG", "UUU", "UUA", "UGG", "UCU"))
```


### Function 5 - 
